### PR TITLE
Accumulate RFI total in timeDownsample

### PR DESCRIPTION
**Pull request by jrs65**
Also, this adds an option to throw away frames that are too old.

### CODE REVIEW

File: lib/stages/timeDownsample.cpp [hunk 1]
```diff
@@ -127,13 +136,21 @@ void timeDownsample::main_thread() {
...
+            output_frame.rfi_total += frame.rfi_total;
 
             // Move to next frame
             nframes += 1;
             mark_frame_empty(in_buf, unique_name.c_str(), frame_id);
             frame_id = (frame_id + 1) % in_buf->num_frames;
 
         } else {
+
+            timespec output_age = std::get<1>(output_frame.time) - std::get<1>(frame.time);
```
Comment (tristpinsm) [inline lib/stages/timeDownsample.cpp]: I may just be confused here, but given we ended up in this block (i.e. `fpga_seq_start > wdw_end`) `frame` is from a later time than `output_frame` so this quantity will always be negative and we will never enter the `if` statement. Should there be a separate `else if` for the case where the frame is too old?

Comment (jrs65) [reply lib/stages/timeDownsample.cpp]: Ooops. Good point. They should be the other way around!

Comment (jrs65) [reply lib/stages/timeDownsample.cpp]: And this is why we do code reviews :)

**Comment by tristpinsm** [comment]
Following up on the timeout case, it means frames that are older than the timeout will get skipped and not trigger the end of the accumulation. Could this potentially lead to a deadlock where the output frame is never marked full?

File: lib/stages/timeDownsample.cpp [hunk 2]
```diff
@@ -127,13 +136,21 @@ void timeDownsample::main_thread() {
...
             nframes += 1;
             mark_frame_empty(in_buf, unique_name.c_str(), frame_id);
             frame_id = (frame_id + 1) % in_buf->num_frames;
 
         } else {
+
+            timespec output_age = std::get<1>(frame.time) - std::get<1>(output_frame.time);
+            if (ts_to_double(output_age) > max_age) {
+                skipped_frame_counter.labels({std::to_string(output_frame.freq_id), "age"}).inc();
+                continue;
```
Comment (andrerenard) [inline lib/stages/timeDownsample.cpp]: I might be missing something but are you missing: 
```
mark_frame_empty(in_buf, unique_name.c_str(), frame_id);
frame_id = (frame_id + 1) % in_buf->num_frames;
```
?

Comment (jrs65) [reply lib/stages/timeDownsample.cpp]: No. The logic is a little confusing, but I don't need that. I did need a `num_frames = 0` though. Or so I've figured out with Tristan.

**Pull request merged**